src/p2p/net_node.inl

```diff
@@ -2001,6 +2066,43 @@ namespace nodetool
     m_peerlist_store_interval.do_call(boost::bind(&node_server<t_payload_net_handler>::store_config, this));
     m_incoming_connections_interval.do_call(boost::bind(&node_server<t_payload_net_handler>::check_incoming_connections, this));
     m_dns_blocklist_interval.do_call(boost::bind(&node_server<t_payload_net_handler>::update_dns_blocklist, this));
+    if (m_exclusive_peers.empty())
+      m_drop_random_outgoing_tor_connection_interval.do_call(boost::bind(&node_server<t_payload_net_handler>::drop_random_outgoing_tor_connection, this));
+    return true;
+  }
+  //-----------------------------------------------------------------------------------
+  template<class t_payload_net_handler>
+  bool node_server<t_payload_net_handler>::drop_random_outgoing_tor_connection()
+  {
+    std::vector<p2p_connection_context> outgoing_tor_connections;
+    auto tor_zone = m_network_zones.find(epee::net_utils::zone::tor);
+
+    if (tor_zone != m_network_zones.end()) {
+        tor_zone->second.m_net_server.get_config_object().foreach_connection([&](p2p_connection_context &cntx) {
+            if (!cntx.m_is_income) {
+                outgoing_tor_connections.push_back(cntx);
+            }
+            return true;
+        });
+    }
+
+    for (auto &cntx : outgoing_tor_connections) {
+      if (!cntx.can_broadcast_txs_to_peer) {
+        drop_connection(cntx);
+        // should use random auth for each connection but no socks5
+        // possible to use same circuit twice if drop initiated by remote
```

Comment:
Cheap/easy way to prevent connecting back until socks5 (mentioned in the comment).

Reply:
But with so few tor peers already, this is going to reduce total connections on the network (because the peer is also running the same logic and may _need_ you as its outbound connection).